src/background/service/keyring/index.ts

```diff
@@ -876,17 +920,30 @@ class KeyringService extends EventEmitter {
    * @returns {Promise<Array<Keyring>>} The keyrings.
    */
   async unlockKeyrings(password: string): Promise<any[]> {
-    // Note that currentAccountIndex is only used in keyring for old accounts that don't have an id stored in the keyring removing in 2.7.6
-    // currentId always takes precedence
-    const currentId = await storage.get('currentId');
     let vaultArray = this.store.getState().vault;
 
     // Ensure vaultArray is an array and filter out null/undefined entries
     vaultArray = Array.isArray(vaultArray) ? vaultArray.filter(Boolean) : [vaultArray];
 
     await this.decryptVaultArray(vaultArray, password);
+
+    if (this.store.getState().vaultVersion !== KEYRING_STATE_VAULT_V2) {
+      await this.encryptVaultArray(this.keyringList, password);
+    }
+
     this.password = password;
```

Comment:
I feel like i ask this everytime i see it, but does this really need to be stored here?